pallets/axelar-router/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+		//
+		//        _execute(sourceChain, sourceAddress, payload);
+		// }
+		#[precompile::public("execute(bytes32,string,string,bytes)")]
+		fn execute(
+			handle: &mut impl PrecompileHandle,
+			_command_id: H256,
+			source_chain: BoundedString<ConstU32<MAX_SOURCE_CHAIN_BYTES>>,
+			source_address: BoundedString<ConstU32<MAX_SOURCE_ADDRESS_BYTES>>,
+			payload: BoundedBytes<ConstU32<MAX_PAYLOAD_BYTES>>,
+		) -> EvmResult {
+			Self::receive(
+				handle.context().caller,
+				source_chain.as_bytes(),
+				source_address.as_bytes(),
+				payload.as_bytes(),
+			)
+			.map_err(|e| TryDispatchError::Substrate(e).into())
+		}
+
+		// Mimics:
+		//
+		//     function executeWithToken(
+		//         bytes32 commandId,
+		//         string calldata sourceChain,
+		//         string calldata sourceAddress,
+		//         bytes calldata payload,
+		//         string calldata tokenSymbol,
+		//         uint256 amount
+		//     ) external { ...
+		//     }
+		//
+		// Note: NOT SUPPORTED
+		//
+		#[precompile::public("executeWithToken(bytes32,string,string,bytes,string,uint256)")]
+		fn execute_with_token(
+			_handle: &mut impl PrecompileHandle,
+			_command_id: H256,
+			_source_chain: BoundedString<ConstU32<MAX_SOURCE_CHAIN_BYTES>>,
+			_source_address: BoundedString<ConstU32<MAX_SOURCE_ADDRESS_BYTES>>,
+			_payload: BoundedBytes<ConstU32<MAX_PAYLOAD_BYTES>>,
+			_token_symbol: BoundedString<ConstU32<MAX_TOKEN_SYMBOL_BYTES>>,
+			_amount: U256,
+		) -> EvmResult {
+			// TODO: Check whether this is enough or if we should error out
+			Ok(())
+		}
+	}
+
+	impl<T: Config> MessageSender for Pallet<T> {
+		type Middleware = AxelarId;
+		type Origin = DomainAddress;
+
+		fn send(axelar_id: AxelarId, origin: Self::Origin, message: Vec<u8>) -> DispatchResult {
+			let chain_name =
+				ChainNameById::<T>::get(axelar_id).ok_or(Error::<T>::RouterNotFound)?;
+			let config = Configuration::<T>::get(&chain_name).ok_or(Error::<T>::RouterNotFound)?;
+
+			match config.domain {
+				DomainConfig::Evm(evm_config) => {
+					let sender_evm_address = H160::from_slice(&origin.address()[0..20]);
+
+					let message = wrap_into_axelar_msg(
+						message,
+						chain_name.into_inner(),
+						config.liquidity_pools_contract_address,
+					)
+					.map_err(DispatchError::Other)?;
+
+					T::Transactor::call(
+						sender_evm_address,
+						evm_config.target_contract_address,
+						message.as_slice(),
+						evm_config.fee_values.value,
+						evm_config.fee_values.gas_price,
+						evm_config.fee_values.gas_limit,
+					)
+					.map(|_| ())
+					.map_err(|e| e.error)
+				}
+			}
+		}
+	}
+}
+
+/// Encodes the provided message into the format required for submitting it
+/// to the Axelar contract which in turn calls the LiquidityPools
+/// contract with the serialized LP message as `payload`.
+///
+/// Axelar contract call:
+/// <https://github.com/axelarnetwork/axelar-cgp-solidity/blob/v4.3.2/contracts/AxelarGateway.sol#L78>
```

Comment:
Nit - maybe we need to update these links? cc @hieronx

Reply:
API still looks the same for the most recent v6.3.1, only the line changed: https://github.com/axelarnetwork/axelar-cgp-solidity/blob/v6.3.1/contracts/AxelarGateway.sol#L150-L156